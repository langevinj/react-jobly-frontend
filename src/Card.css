.Card-body {
    display: flex;
    box-sizing: border-box;
    flex: 1 1 auto;
    max-height: 250px;
    background-color: tan;
    margin-bottom: 1rem;
}