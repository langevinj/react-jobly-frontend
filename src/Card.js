import React from 'react'
import CompanyCard from './CompanyCard'
import JobCard from './JobCard'
import { Link } from 'react-router-dom'
import './Card.css'

function Card({ title, item, apply = () => null, idx }){

    if(title === 'companies'){
        return (
            <Link to={`/companies/${item.handle}`}>
                <div className="Card-body"><CompanyCard company={item} /></div>
            </Link>
        )
    } else {
        return (
            <div className="Card-body">
                <JobCard job={item} handleApply={() => apply(idx)} />
            </div>
        )
    }
    
}

export default Card;