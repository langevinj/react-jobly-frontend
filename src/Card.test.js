import React from 'react'
import { render } from '@testing-library/react'
import Card from './Card'
import { MemoryRouter } from "react-router"

it('renders without crashing', function () {
    render(
    <MemoryRouter>
            <Card />
        </MemoryRouter>);
});

it("matches the snapshot for a company", function() {
    let item = {
        handle: "test-company",
        name: "Test Company",
        description: "a test company"
    };

    const { asFragment } = render(
        <MemoryRouter>
            <Card item={item}/>
        </MemoryRouter>
    )

    expect(asFragment()).toMatchSnapshot();
});

it("matches the snapshot for a job", function() {
    let item = {
        title: "test job",
        salary: 50000,
        equity: 0.4
    }

    const { asFragment } = render(
        <MemoryRouter>
            <Card item={item} />
        </MemoryRouter>
    )

    expect(asFragment()).toMatchSnapshot();
});