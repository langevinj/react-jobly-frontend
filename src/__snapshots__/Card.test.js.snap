// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot for a company 1`] = `
<DocumentFragment>
  <div
    class="Card-body"
  >
    <div
      class="JobCard"
    >
      <h6 />
      <p>
        Salary: 
      </p>
      <p>
        Equity: 
      </p>
      <button
        class="Apply btn-primary rounded"
        data-toggle="button"
        type="button"
      >
        Apply
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches the snapshot for a job 1`] = `
<DocumentFragment>
  <div
    class="Card-body"
  >
    <div
      class="JobCard"
    >
      <h6>
        test job
      </h6>
      <p>
        Salary: 50000
      </p>
      <p>
        Equity: 0.4
      </p>
      <button
        class="Apply btn-primary rounded"
        data-toggle="button"
        type="button"
      >
        Apply
      </button>
    </div>
  </div>
</DocumentFragment>
`;
